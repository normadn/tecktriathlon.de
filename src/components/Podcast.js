import React from "react";

import ActionButton from "./ActionButton";
import ContentWrapper from "./ContentWrapper";

import circuits from "../assets/img/backgrounds/circuitsGrey.png";
import podcastwebp from "../assets/img/icons/podcastIcon.webp";
import podcastpng from "../assets/img/icons/getInTouch.png";

const Podcast = () => (
  <section id="kontakt">
    <div
      css={{
        paddingBottom: "133px",
        width: "100%",
        position: "relative"
      }}
    >
      <ContentWrapper>
        <div
          css={{
            position: "relative",
            padding: "60px 72px 60px 72px",
            backgroundColor: "#ffffff",
            borderRadius: "8px",
            boxShadow: "4px 9px 40px 2px rgba(7, 16, 75, 0.1)",
            border: "solid 1px #e8e8e8",
            display: "grid",
            gridTemplateColumns: "min-content auto",
            "@media (max-width: 1128px)": { display: "block" },
            padding: "40px 5% 40px 5%"
          }}
        >
          <div
            css={{
              "@media (max-width: 1128px)": { textAlign: "center" }
            }}
          >
            <picture
              css={{
                "@media (max-width: 1128px)": {
                  width: "100%"
                },
                maxWidth: "290px"
              }}
            >
              <source srcSet={podcastpng} type="image/png" />
              <img
                css={{
                  "@media (max-width: 1128px)": {
                    width: "100%"
                  },
                  maxWidth: "290px"
                }}
                src={podcastpng}
                alt="Headphones connected to phone"
              />
            </picture>
          </div>
          <div
            css={{
              textAlign: "left",
              paddingLeft: "64px",
              "@media (max-width: 1128px)": {
                paddingLeft: "0px",
                paddingTop: "46px",
                textAlign: "center"
              }
            }}
          >
            <form name="contact" method="POST" data-netlify="true">
              <h1 css={{ paddingBottom: "24px", paddingTop: "10px" }}>
                Kontaktanfrage
              </h1>

              <input
                placeholder="Ihr Name"
                type="text"
                name="name"
                css={{
                  width: "80%",
                  flexGrow: 2,
                  padding: "0 16px",
                  height: "48px",
                  lineHeight: "48px",
                  borderRadius: "4px",
                  fontSize: "16px",
                  marginBottom: "15px"
                }}
                aria-label="Ihr Name"
              />

              <input
                placeholder="Ihre Email Addresse"
                type="email"
                name="email"
                css={{
                  width: "80%",
                  flexGrow: 2,
                  padding: "0 16px",
                  height: "48px",
                  lineHeight: "48px",
                  borderRadius: "4px",
                  fontSize: "16px",
                  marginBottom: "15px"
                }}
                aria-label="Ihre Email Addresse"
              />

              <div
                css={{
                  width: "100%",
                  "@media screen and (max-width: 1128px)": {
                    display: "flex",
                    alignItems: "center",
                    justifyContent: "center"
                  }
                }}
              >
                <ActionButton
                  type="submit"
                  foregroundColor="#00205b"
                  backgroundColor="#ffffff"
                >
                  Senden
                </ActionButton>
              </div>
            </form>
          </div>
        </div>
      </ContentWrapper>
    </div>
    <div
      css={{
        marginTop: "-315px",
        height: "315px",
        zIndex: "-1",
        position: "relative",
        background: `url(${circuits}) no-repeat center center`,
        backgroundSize: "cover",
        borderTop: "solid #dbdbdb 3px"
      }}
    />
  </section>
);

export default Podcast;
