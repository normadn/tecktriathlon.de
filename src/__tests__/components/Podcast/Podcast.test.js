import React from "react";
import Podcast from "../../../components/Podcast";
import { shallow } from "enzyme";

describe("Podcast", () => {
  it("renders correctly", () => {
    expect(shallow(<Podcast />)).toMatchSnapshot();
  });
});
