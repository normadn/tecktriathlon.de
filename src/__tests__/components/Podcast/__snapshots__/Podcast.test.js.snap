// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Podcast renders correctly 1`] = `
<section>
  <div
    css={
      Object {
        "paddingBottom": "133px",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <_default>
      <div
        css={
          Object {
            "@media (max-width: 1128px)": Object {
              "display": "block",
            },
            "backgroundColor": "#ffffff",
            "border": "solid 1px #e8e8e8",
            "borderRadius": "8px",
            "boxShadow": "4px 9px 40px 2px rgba(7, 16, 75, 0.1)",
            "display": "grid",
            "gridTemplateColumns": "min-content auto",
            "padding": "40px 5% 40px 5%",
            "position": "relative",
          }
        }
      >
        <div
          css={
            Object {
              "@media (max-width: 1128px)": Object {
                "textAlign": "center",
              },
            }
          }
        >
          <picture
            css={
              Object {
                "@media (max-width: 1128px)": Object {
                  "width": "100%",
                },
                "maxWidth": "290px",
              }
            }
          >
            <source
              srcSet="test-file-stub"
              type="image/webp"
            />
            <source
              srcSet="test-file-stub"
              type="image/png"
            />
            <img
              alt="Headphones connected to phone"
              css={
                Object {
                  "@media (max-width: 1128px)": Object {
                    "width": "100%",
                  },
                  "maxWidth": "290px",
                }
              }
              src="test-file-stub"
            />
          </picture>
        </div>
        <div
          css={
            Object {
              "@media (max-width: 1128px)": Object {
                "paddingLeft": "0px",
                "paddingTop": "46px",
                "textAlign": "center",
              },
              "paddingLeft": "64px",
              "textAlign": "left",
            }
          }
        >
          <h1
            css={
              Object {
                "paddingBottom": "24px",
                "paddingTop": "10px",
              }
            }
          >
            Listen to our podcast!
          </h1>
          <p
            css={
              Object {
                "fontWeight": "500",
                "lineHeight": "1.5",
                "paddingBottom": "40px",
              }
            }
          >
            Queen’s University has produced a lot of great tech talent - and we want to hear their stories! The QHacks Podcast is our opportunity to learn from current and former Queen’s students who have become leaders, innovators, and entrepreneurs in the tech industry!
          </p>
          <div
            css={
              Object {
                "@media screen and (max-width: 1128px)": Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                },
                "width": "100%",
              }
            }
          >
            <ActionButton
              backgroundColor="#ffffff"
              dataCy="listen-button"
              foregroundColor="#00205b"
              link="https://open.spotify.com/show/6eZg09yz4xKgynJdZeDmo1"
              type="rounded"
            >
              Listen
            </ActionButton>
          </div>
        </div>
      </div>
    </_default>
  </div>
  <div
    css={
      Object {
        "background": "url(test-file-stub) no-repeat center center",
        "backgroundSize": "cover",
        "borderTop": "solid #dbdbdb 3px",
        "height": "315px",
        "marginTop": "-315px",
        "position": "relative",
        "zIndex": "-1",
      }
    }
  />
</section>
`;
